src/platform/Tizen/ConnectivityManagerImpl.cpp

```diff
@@ -69,68 +81,162 @@ bool ConnectivityManagerImpl::_HaveIPv6InternetConnectivity(void)
 
 ConnectivityManager::WiFiStationMode ConnectivityManagerImpl::_GetWiFiStationMode(void)
 {
-    return ConnectivityManager::kWiFiStationMode_NotSupported;
+    CHIP_ERROR err                          = CHIP_NO_ERROR;
+    wifi_manager_device_state_e deviceState = WIFI_MANAGER_DEVICE_STATE_DEACTIVATED;
+
+    if (mWiFiStationMode != kWiFiStationMode_ApplicationControlled)
+    {
+        err = WiFiMgr().GetDeviceState(&deviceState);
+        SuccessOrExit(err);
+
+        if (deviceState == WIFI_MANAGER_DEVICE_STATE_ACTIVATED)
+        {
+            mWiFiStationMode = kWiFiStationMode_Enabled;
+        }
+        else
+        {
+            mWiFiStationMode = kWiFiStationMode_Disabled;
+        }
+    }
+
+exit:
```

Comment:
exit has no cleanup section. Could we use return macros instead? Same comment for several methods below.

```cpp

ConnectivityManager::WiFiStationMode ConnectivityManagerImpl::_GetWiFiStationMode(void)
{
    ReturnErrorCodeIf(mWiFiStationMode == kWiFiStationMode_ApplicationControlled, CHIP_NO_ERROR);
    
    wifi_manager_device_state_e deviceState = WIFI_MANAGER_DEVICE_STATE_DEACTIVATED;
    ReturnErrorOnFailure(WiFiMgr().GetDeviceState(&deviceState));
    
    mWifiStationMode = 
        (deviceState == WIFI_MANAGER_DEVICE_STATE_ACTIVATED)
            ? kWiFiStationMode_Enabled 
            : kWiFiStationMode_Disabled;

    return CHIP_NO_ERROR;
}

```

Reply:
Thanks for pointing out. Will update.